### PR TITLE
Printing only xpath that not matches

### PR DESCRIPTION
**Pull request by jrdalpra**
Solving issue #26 

- New matcher that check if a collection of matchers matches and print info only the ones that don't matches
- Changing the of "Matchers.allOf" for this new one
- Added a new test case to check if AssertionError has info only about the wrong xpaths

### CODE REVIEW

**Comment by dmarkov** [comment]
@jrdalpra Many thanks for the PR, let me find a reviewer for it

**Comment by dmarkov** [comment]
@dmzaytsev please review

File: src/main/java/com/jcabi/matchers/AllOfThatPrintsOnlyWrongMatchers.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    private final transient List<Matcher<? super T>> wrong;
+    /**
+     * Construct that accept matchers to test.
+     * @param iterable Matchers that will be tested.
+     */
+    public AllOfThatPrintsOnlyWrongMatchers(
+            final Iterable<Matcher<? super T>> iterable) {
+        super();
+        this.matchers = iterable;
+        this.wrong = new ArrayList<Matcher<? super T>>(SIZE);
```
Comment (dmzaytsev) [inline src/main/java/com/jcabi/matchers/AllOfThatPrintsOnlyWrongMatchers.java]: @jrdalpra you can use `Tv.Ten` instead
But I think even `1` will be enough

File: src/main/java/com/jcabi/matchers/AllOfThatPrintsOnlyWrongMatchers.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        boolean matches = true;
+        for (final Matcher<? super T> matcher : this.matchers) {
+            if (!matcher.matches(obj)) {
+                mismatch.appendDescriptionOf(matcher).appendText(" ");
+                matcher.describeMismatch(obj, mismatch);
+                this.wrong.add(matcher);
+                matches = false;
+            }
+        }
+        return matches;
```
Comment (dmzaytsev) [inline src/main/java/com/jcabi/matchers/AllOfThatPrintsOnlyWrongMatchers.java]: @jrdalpra let's say you have three `Matcher`  that return
`true`
`false`
`true`

what will be returned in this case?

File: src/test/java/com/jcabi/matchers/XhtmlMatchersTest.java [hunk 3]
```diff
@@ -237,6 +237,36 @@ public void hasXPaths() throws Exception {
...
+     * @throws Exception
+     */
+    @Test
+    public void hasXPathsPrintsOnlyWrongXPaths() {
+        try {
+            MatcherAssert.assertThat(
+                "<b><file>some.txt</file><file>ghi.txt</file></b>",
+                XhtmlMatchers.hasXPaths(
+                    "/b/file[.='some.txt']",
+                    "/b/file[.='ghx.txt']"
```
Comment (dmzaytsev) [inline src/test/java/com/jcabi/matchers/XhtmlMatchersTest.java]: @jrdalpra let's add one more case for `"/b/file[.='gni.txt']"` after this line

**Comment by dmzaytsev** [comment]
@jrdalpra please see my comments

File: src/main/java/com/jcabi/matchers/AllOfThatPrintsOnlyWrongMatchers.java (hunk 2 above)
Comment (jrdalpra) [inline src/main/java/com/jcabi/matchers/AllOfThatPrintsOnlyWrongMatchers.java]: @dmzaytsev only xpath that NOT matches

File: src/test/java/com/jcabi/matchers/XhtmlMatchersTest.java (hunk 3 above)
Comment (jrdalpra) [inline src/test/java/com/jcabi/matchers/XhtmlMatchersTest.java]: @dmzaytsev done

File: src/main/java/com/jcabi/matchers/AllOfThatPrintsOnlyWrongMatchers.java (hunk 1 above)
Comment (jrdalpra) [inline src/main/java/com/jcabi/matchers/AllOfThatPrintsOnlyWrongMatchers.java]: @dmzaytsev done. The minimum is THREE.

**Comment by dmzaytsev** [comment]
@jrdalpra thanks

**Comment by dmzaytsev** [comment]
@rultor merge

**Comment by jrdalpra** [comment]
@yegor256 can you confirm, please?

**Comment by yegor256** [comment]
@rultor merge

**Comment by yegor256** [comment]
@rultor merge

**Comment by dmarkov** [comment]
@dmzaytsev **21 mins** added to your account (payment number `75991578`), many thanks for your contribution! 98 hours and 17 mins spent here.; you're getting extra minutes here (c=6); +21 to your rating, your total score is [+2909](http://www.netbout.com/b/35166?open=rating)

**Comment by dmarkov** [comment]
@rultor deploy

**Pull request merged**